osf/models/mixins.py

```diff
@@ -773,15 +809,32 @@ def set_subjects(self, new_subjects, auth, add_log=True):
                     self.subjects.add(Subject.load(s_id))
 
         if add_log and hasattr(self, 'add_log'):
-            self.add_log(
-                action=NodeLog.SUBJECTS_UPDATED,
-                params={
-                    'subjects': list(self.subjects.values('_id', 'text')),
-                    'old_subjects': list(Subject.objects.filter(id__in=old_subjects).values('_id', 'text'))
-                },
-                auth=auth,
-                save=False,
-            )
+            self.add_subjects_log(old_subjects, auth)
+
+        self.save(old_subjects=old_subjects)
+
+    def set_subjects_from_relationships(self, subjects_list, auth, add_log=True):
+        """ Helper for setting M2M subjects field from list of flattened subjects received from UI.
+        Only authorized admins may set subjects.
+
+        :param list[Subject._id] new_subjects: List of flattened subject hierarchies
+        :param Auth auth: Auth object for requesting user
+        :param bool add_log: Whether or not to add a log (if called on a Loggable object)
+
+        :return: None
+        """
```

Comment:
This method is called when subjects are serialized under relationships. Will just be a flattened list of subjects.